### ISSUE TITLE
Export xyBase

### ISSUE THREAD

**Issue by ColinEberhardt** [d3fc/d3fc#1120]
_From @onedayitwillmake on April 28, 2018 7:55_

Great library!

It would be great if `xyBase` was also exported (via the `build/d3fc` bundle), currently it's not possible to create a custom series due to this minor limitation. 

_Copied from original issue: d3fc/d3fc-series#56_

**Comment by ColinEberhardt** [comment]
_From @chrisprice on April 28, 2018 8:14_

Thanks Mario. We don't export the base components because we consider their
API to be an internal implementation detail. Having said that, you probably
don't need the functionality they contain to create your own series.

Most of the code contained within them is just to allow the series to be
fully customisable. If you're making a bespoke series you probably don't
need to offer that much flexibility.

The only requirements we impose on a series component in order for it to be
compatible with the library is for it to accept an xScale and a yScale.

**Comment by ColinEberhardt** [comment]
_From @onedayitwillmake on April 28, 2018 21:22_

As an example, I was attempting to make a discontinuous `svgLineSeries`, however the `defined` property due to how it is bound - is cannot be overwritten.

in xyBase.js
```javascript
    base.defined = (d, i) => defined(baseValue, crossValue, mainValue)(d, i);
```

Attempting to set the property via `fc.svgLineSeries().defined(MyMethod)` simply calls the method outlined above. (Perhaps this is a bug? I could not discern the intent, so I carried on). Instead I tried to simply expose the `lineData` which is not exposed by `svgLineSeries`, which I could then call `define(MyCustomMethod)` passing my own function which would return a truthy value.

I do not see harm in exposing the `xyBase` class, it allows someone to create their own custom series while following the conventions in `d3fc` so their custom series is familiar to anyone else consuming it.
